test/api/small-catches.unit.test.js

```diff
@@ -0,0 +1,261 @@
+
+      const smallCatchesApi = new SmallCatchesApi()
+      await smallCatchesApi.add(request, 'sub-1', 'act-1', 4, [{ c: 1 }], true, false)
+
+      expect(superAddMock).toHaveBeenCalledWith(
+        request,
+        expect.objectContaining({
+          submission: 'sub-1',
+          activity: 'act-1',
+          month: 'APR',
+          released: true,
+          counts: [{ c: 1 }],
+          noMonthRecorded: false
+        })
+      )
+    })
+  })
+
+  describe('change', () => {
+    it('returns result immediately when super.change returns errors', async () => {
+      EntityApi.prototype.change = jest.fn().mockResolvedValueOnce({
+        errors: ['something bad']
+      })
+
+      const smallCatchesApi = new SmallCatchesApi()
+      const result = await smallCatchesApi.change({}, 'C1', 'A1', 4, [], false, false)
+
+      expect(result).toEqual({ errors: ['something bad'] })
+    })
+
+    it('changes association if mapped activity differs', async () => {
+      EntityApi.prototype.change = jest.fn().mockResolvedValueOnce({ id: 'C1' })
+      const request = {}
+      EntityApi.prototype.changeAssoc = jest.fn().mockResolvedValueOnce({})
+      const smallCatchesApi = new SmallCatchesApi()
+      smallCatchesApi.doMap = jest.fn().mockResolvedValueOnce({
+        activity: { id: 'A2' }
+      })
+
+      await smallCatchesApi.change(request, 'C1', 'A1', 9, [], false, false)
+
+      expect(EntityApi.prototype.changeAssoc).toHaveBeenCalledWith(request, 'C1/activity', 'A1')
+    })
+
+    it('returns duplicate error on conflict when changing association', async () => {
+      EntityApi.prototype.change = jest.fn().mockResolvedValueOnce({ id: 'C1' })
+      EntityApi.prototype.changeAssoc = jest.fn().mockResolvedValueOnce({ statusCode: ResponseError.status.CONFLICT })
+      const smallCatchesApi = new SmallCatchesApi()
+      smallCatchesApi.doMap = jest.fn().mockResolvedValueOnce({
+        activity: { id: 'A2' }
+      })
+
+      const result = await smallCatchesApi.change({}, 'C1', 'A1', 9, [], false, false)
+
+      expect(result).toEqual({
+        errors: [
+          {
+            entity: 'SmallCatch',
+            message: 'SMALL_CATCH_DUPLICATE_FOUND'
+          }
+        ]
+      })
+    })
+  })
+
+  describe('changeExclusion', () => {
+    it('calls super.change with reportingExclude', async () => {
+      EntityApi.prototype.change = jest.fn()
+      const request = {}
+
+      const smallCatchesApi = new SmallCatchesApi()
+      await smallCatchesApi.changeExclusion(request, 'C1', true)
+
+      expect(EntityApi.prototype.change).toHaveBeenCalledWith(
+        request,
+        'C1',
+        { reportingExclude: true }
+      )
+    })
+  })
+
+  describe('sort', () => {
+    const buildCatch = (overrides = {}) => ({
+      month: 6,
+      noMonthRecorded: false,
+      activity: {
+        river: { name: 'Default River' },
+        ...(overrides.activity || {})
+      },
+      ...overrides
+    })
```

Comment:
The bit where you're using the spread operator on `overrides.activity` will never actually result in anything, because it'll be over-ridden by the spread operator on `overrides` just below. For example, if you do the following:
```    
    console.log(buildCatch({ activity: { abc: 123 } }))
```
You'll see this:
```
{
  "month": 6,
  "noMonthRecorded": false,
  "activity": {
    "abc": 123
  }
}
```
If the `overrides.activity` was honoured, then the activity key would point at the object `{ river: { name: 'Default River' } , 'abc': 123 }`. This may be causing problems in your tests...